src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -359,4 +360,95 @@ public void registerUserForCourse(User user, Course course) {
         auditEventRepository.add(auditEvent);
         log.info("User " + user.getLogin() + " has successfully registered for course " + course.getTitle());
     }
+
+    /**
+     * Get the active students for this particular course
+     *
+     * @param courseId the id of the course
+     * @param periodIndex an index indicating which time period, 0 is current week, -1 is one week in the past, -2 is two weeks in the past ...
+     */
+    public Integer[] getCourseStatistics(Long courseId, Integer periodIndex) {
+        Integer[] result = new Integer[12];
```

Comment:
Is there a particular reason why you're creating the array here and not in `createResultArray`? It seems that `result` isn't really used in this method (apart from filling it with 0).

Reply:
Thats a good catch, i think i can postpone this call so i do not need to take this as a parameter